src/azul/indexer/index_controller.py

```diff
@@ -158,7 +171,13 @@ def _validate_notification(self, notification):
     def contribute(self, event: Iterable[SQSRecord], *, retry=False):
         for record in event:
             message = json.loads(record.body)
-            attempts = record.to_dict()['attributes']['ApproximateReceiveCount']
+            attempts = message['attempts']
+            if attempts > 1 and not retry:
+                log.warning('Non-retry worker skipping message: %r', message)
+                raise ContributeLambdaError(f'Worker already attempted message: {message!r}')
```

Comment:
Call change message visibility timeout client to set visibility to 0 and immediately return the message to the retry lambda.